src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -795,6 +795,11 @@ private void DetectAttributeAndOptionConflicts(DiagnosticBag diagnostics)
                 }
             }
 
+            if (_compilation.Options.PublicSign && (object)this.AssemblyKeyContainerAttributeSetting != (object)CommonAssemblyWellKnownAttributeData.StringMissingValue)
+            {
+                diagnostics.Add(ErrorCode.WRN_AttributeIgnoredWhenPublicSigning, NoLocation.Singleton, AttributeDescription.AssemblyKeyNameAttribute.FullName);
```

Comment:
Yes, they're the same thing. The meaning is I think sort of "the name of the container", so it's a bit strange. Here's the point at which the attribute (KeyName) gets assigned to the setting (KeyContainer): https://github.com/dotnet/roslyn/blob/96f5ce2f5461b1822c96b714ac48e5a380161c8d/src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs#L2116

However, it is *NOT* the same as KeyFile, which is a separate check/diagnostic below this one.